jlm/llvm/opt/alias-analyses/RegionAwareMemoryNodeProvider.hpp

```diff
@@ -210,7 +191,19 @@ class RegionAwareMemoryNodeProvider final : public MemoryNodeProvider
   static std::string
   ToRegionTree(const rvsdg::Graph & rvsdg, const RegionAwareMemoryNodeProvisioning & provisioning);
 
+  // The provisioning produced
   std::unique_ptr<RegionAwareMemoryNodeProvisioning> Provisioning_;
+
+  // The SCCs of functions, in reverse topological order
+  // If function a() calls b(), and they are not in the same SCC, a()'s SCC comes after b()'s SCC.
+  std::vector<util::HashSet<const rvsdg::LambdaNode *>> FunctionSCCs_;
+  // Which SCC contains the node representing external functions
+  size_t ExternalNodeSccIndex_;
+  // A mapping from functions to the index of the SCC they belong to in the call graph
+  std::unordered_map<const rvsdg::LambdaNode *, size_t> FunctionToSccIndex_;
+
+  // The set of memory nodes that may be affected within a given SCC. Indexed by sccIndex.
+  std::vector<util::HashSet<const PointsToGraph::MemoryNode *>> SccSummaries_;
```

Comment:
Could we get some spaces between the individual variables and transform the comments to doxygen comments?